Automattic/wp-calypso
# Display Razorpay icon in saved payment methods

Closes: https://github.com/Automattic/payments-florin/issues/435

## Proposed Changes

In adding support for Razorpay, we need to improve how saved payment methods are displayed to customers.  Currently, a placeholder credit card icon and `razorpay` are what visitors see.  This PR changes that by replacing the icon with the Razorpay logo and having `Razorpay` displayed if an email address is not available.

**Before**
![razorpay-before](https://github.com/Automattic/wp-calypso/assets/1138631/fceab9fc-245d-46f5-af9f-cf1586394e77)

**After**
![razorpay-payment-method](https://github.com/Automattic/wp-calypso/assets/1138631/d6239387-48da-4161-899f-3633c442217d)


## Testing Instructions

Taken from https://github.com/Automattic/wp-calypso/pull/86197

* (a12s) Enable store sandbox and billing sandbox.
* Change your user's currency to INR.
* In checkout, set your tax location to India. Use any 6-digit numeric postal code.
* After clicking to submit in checkout, a modal window will appear.
 * If prompted to enter a phone number, use 7513200000.
 * You will be presented with a list of payment methods. Docs about testing these is [here](https://razorpay.com/docs/payments/payment-gateway/ecommerce-plugins/wordpress/test-integration/).
 * Card: Use 4111 1111 1111 1111 with any future expiration date, any 3 digit cvv, and any 4 digit 2FA code to simulate a successful payment.
* Once payment has been successful, go to `me/purchases/payment-methods` and you should see the Razorpay logog next to the payment method.

## Pre-merge Checklist
